### PR TITLE
prevent integer values overflowing our index datatypes

### PR DESCRIPTION
**Pull request by Dieterbe**
the cassandra index uses 'int' type which is a 32bit signed integer.
If we allow extremely high interval and timestamp values then
we will get errors trying to update index in cassandra, which will
keep retrying and blocking metrictank.

### CODE REVIEW

File: input/input.go [hunk 1]
```diff
@@ -62,7 +63,9 @@ func (in DefaultHandler) ProcessMetricPoint(point schema.MetricPoint, format msg
 	} else {
 		in.receivedMPNO.Inc()
 	}
-	if !point.Valid() {
+	// in cassandra we store timestamps as 32bit signed integers.
+	// math.MaxInt32 = Jan 19 03:14:07 UTC 2038
+	if !point.Valid() || point.Time >= math.MaxInt32 {
```
Comment (replay) [inline input/input.go]: is this the same for bigtable?

File: input/input.go [hunk 2]
```diff
@@ -89,7 +92,9 @@ func (in DefaultHandler) ProcessMetricData(md *schema.MetricData, partition int3
 		log.Debugf("in: Invalid metric %v: %s", md, err)
 		return
 	}
-	if md.Time == 0 {
+	// in cassandra we store timestamps and interval as 32bit signed integers.
+	// math.MaxInt32 = Jan 19 03:14:07 UTC 2038
+	if md.Time == 0 || md.Time >= math.MaxInt32 || md.Interval >= math.MaxInt32 {
```
Comment (DanCech) [inline input/input.go]: log message needs to be updated

Comment (replay) [inline input/input.go]: I'm assuming that Time & Interval will only be `>math.MaxInt32` if the received data is somehow corrupted. If we're now filtering corrupt data here we should probably also check if they are `<0`

File: input/input.go (hunk 1 above)
Comment (Dieterbe) [reply input/input.go]: in bigtable we use varint encoding and have 8bytes to work with.
I don't have a strong opinion on where this should go. the alternative is we add 2 interface methods to the index plugins, which for bigtable just return true, and in cassandra's case, the validation logic gets split into two different locations. is that an advantegous design? i don't know.

the benefit of using the lowest common denominator is also that it allows data to be moved from one store to another

File: input/input.go (hunk 2 above)
Comment (Dieterbe) [reply input/input.go]: good point, but that should be added to the schema repository.
the validation here is only for additional validation due to MT's limitations

File: input/input.go (hunk 1 above)
Comment (DanCech) [reply input/input.go]: We can always do that later if it becomes an issue, but I think this is fine.

File: input/input.go (hunk 2 above)
Comment (DanCech) [reply input/input.go]: log message still needs to be updated

Comment (DanCech) [reply input/input.go]: I'd still change from `== 0` to `<= 0` here on the off chance something strange happens (and log the received value).

**Pull request merged**